src-extras/Database/LSMTree/Extras/Orphans.hs

```diff
@@ -25,13 +39,105 @@ import           System.Random (Uniform)
 deriving anyclass instance Uniform Word256
 
 instance SerialiseKey Word256 where
-  serialiseKey (Word256{word256hi, word256m1, word256m0, word256lo}) =
+  serialiseKey w256 =
     RB.RawBytes $ mkPrimVector 0 32 $ P.runByteArray $ do
       ba <- P.newByteArray 32
-      P.writeByteArray ba 0 $ byteSwap64 word256hi
-      P.writeByteArray ba 1 $ byteSwap64 word256m1
-      P.writeByteArray ba 2 $ byteSwap64 word256m0
-      P.writeByteArray ba 3 $ byteSwap64 word256lo
+      P.writeByteArray ba 0 $ byteSwapWord256 w256
       return ba
+  deserialiseKey (RawBytes (PV.Vector off len ba))
+    | len >= expected = byteSwapWord256 $ indexWord8ArrayAsWord256 ba off
+    | otherwise = error $ deserialiseKeyInputBytesNotPrecise "Word256" len expected
+    where expected = 32
 
-  deserialiseKey = error "deserialiseKey: Word256" -- TODO
+instance SerialiseValue Word256 where
+  serialiseValue w256 =
+    RB.RawBytes $ mkPrimVector 0 32 $ P.runByteArray $ do
+      ba <- P.newByteArray 32
+      P.writeByteArray ba 0 w256
+      return ba
+  deserialiseValue (RawBytes (PV.Vector off len ba))
+    | len >= expected = indexWord8ArrayAsWord256 ba off
+    | otherwise = error $ deserialiseValueInputBytesNotPrecise "Word256" len expected
+    where expected = 32
```

Comment:
I though you wanted to check that the length exactly everywhere (https://github.com/IntersectMBO/lsm-tree/pull/185#discussion_r1583193899), not just just have a lower bound? But that's only done for the `UTxO` key.
Or is it because that instance is in `Extras`, but the core instances should rather not fail? If so, at least an `assert` could make sense, perhaps even including the check that `ba` is large enough (basically `isValidSlice`).

If the usage of the new helper function is very similar for all instances, one might even think about also factoring out the `len == expected` part and just using it like this:
```haskell
  deserialiseValue (RawBytes (PV.Vector off len ba)) =
      requireBytesExactly 32 "Word256" len $
        indexWord8ArrayAsWord256 ba off
```

But might be too much.